api/v1/clusterextensionrevision_types.go

```diff
@@ -22,7 +22,24 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-const ClusterExtensionRevisionKind = "ClusterExtensionRevision"
+const (
+	ClusterExtensionRevisionKind = "ClusterExtensionRevision"
+
+	// Condition Types
+	TypeAvailable = "Available"
+	TypeSucceeded = "Succeeded"
+
+	// Condition Reasons
+	ReasonObjectAvailable           = "Available"
+	ReasonReconcileFailure          = "ReconcileFailure"
+	ReasonRevisionValidationFailure = "RevisionValidationFailure"
+	ReasonPhaseValidationError      = "PhaseValidationError"
+	ReasonObjectCollisions          = "ObjectCollisions"
+	ReasonRolloutSuccess            = "RolloutSuccess"
+	ReasonProbeFailure              = "ProbeFailure"
+	ReasonIncomplete                = "Incomplete"
+	ReasonProgressing               = "Progressing"
```

Comment:
I know this PR faithfully creates constants for everything that we're actually using, and it is almost the point of this PR to centralize this and make it plainly obvious what our true surface is. So this is what I expect of this PR! :)

So now that we see this, here are meta questions meant for all of us:
1. do we need this many reasons?
2. are these the "right" reasons? (for example, I added "ReconcileFailure" on a whim just to hack the status plumbing together)